link_test.go

```diff
@@ -664,3 +667,151 @@ func TestLinkStats64UnmarshalBinary(t *testing.T) {
 		})
 	}
 }
+
+func TestLinkXDP(t *testing.T) {
+	// BPF compilation requires a high RLIMIT_MEMLOCK
+	n := uint64(1024 * 1024 * 10)
+	err := unix.Setrlimit(unix.RLIMIT_MEMLOCK, &unix.Rlimit{Cur: n, Max: n})
+	if err != nil {
+		t.Fatalf("failed to increase RLIMIT_MEMLOCK: %s", err)
+	}
+
+	// establish a netlink connections
+	conn, err := Dial(nil)
+	if err != nil {
+		t.Fatalf("failed to establish netlink socket")
+	}
+	defer conn.Close()
+
+	// construct dummy interface to test XDP program against
+	if err := conn.Link.New(&LinkMessage{
+		Family: unix.AF_UNSPEC,
+		Index:  1001,
+		Flags:  unix.IFF_UP,
+		Attributes: &LinkAttributes{
+			Name: "dummy0",
+			Info: &LinkInfo{Kind: "dummy"},
+		},
+	}); err != nil {
+		t.Fatalf("failed to construct dummy interface: %s", err)
+	}
+
+	// get info for the dummy interface
+	interf, err := conn.Link.Get(1001)
+	if err != nil {
+		conn.Link.Delete(interf.Index)
+		t.Fatalf("could not get dummy interface info: %s", err)
+	}
+
+	// compile a BPF test program. If it fails error out of the tests
+	// and clean up dummy interface.
+	prog, err := ebpf.NewProgram(&ebpf.ProgramSpec{
+		Type: ebpf.XDP,
+		Instructions: asm.Instructions{
+			asm.LoadImm(asm.R0, int64(2), asm.DWord),
```

Comment:
Can you leave a quick comment to explain what exactly this does? I assume it's more or less a noop?

Reply:
It's just loading `2` ([XDP_PASS](https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/bpf.h#L4366)) into the [return values register](https://pkg.go.dev/github.com/cilium/ebpf/asm#Register) of the BPF program, doing essentially nothing with the packet, so a noop